src/herder/TxSetFrame.cpp

```diff
@@ -336,6 +369,14 @@ TxSetFrame::makeFromTransactions(TxPhases const& txPhases, Application& app,
         txSet->toXDR(xdrTxSet);
         outputTxSet = TxSetFrame::makeFromWire(app, xdrTxSet);
     }
+#ifdef BUILD_TESTS
+    if (skipValidation)
+    {
+        txSet->mHash = outputTxSet->mHash;
```

Comment:
why not `return outputTxSet;`?

Reply:
This is needed to not use the new tx frame objects, otherwise the apply order override wouldn't match the pointers stored in tx set (and we do need them to match in order for fee lookups to work).

Reply:
Oh I see, makes sense, thanks